astrobase/periodbase/tls.py

```diff
@@ -0,0 +1,380 @@
+        time-series (with non-finite elems removed) will be passed through to
+        the output.
+
+    verbose : bool
+        Kept for consistency with `periodbase` functions.
+
+    nworkers : int or None
+        The number of parallel workers to launch for period-search. If None,
+        nworkers = NCPUS.
+
+    Returns
+    -------
+
+    dict
+        This function returns a dict, referred to as an `lspinfo` dict in other
+        astrobase functions that operate on periodogram results. The format is
+        similar to the other astrobase period-finders -- it contains the
+        nbestpeaks, which is the most important thing. (But isn't entirely
+        standardized.)
+
+        Crucially, it also contains "tlsresult", which is a dictionary with
+        transitleastsquares spectra (used to get the SDE as defined in the TLS
+        paper), statistics, transit period, mid-time, duration, depth, SNR, and
+        the "odd_even_mismatch" statistic. The full key list is:
+
+            dict_keys(['SDE', 'SDE_raw', 'chi2_min', 'chi2red_min', 'period',
+            'period_uncertainty', 'T0', 'duration', 'depth', 'depth_mean',
+            'depth_mean_even', 'depth_mean_odd', 'transit_depths',
+            'transit_depths_uncertainties', 'rp_rs', 'snr', 'snr_per_transit',
+            'snr_pink_per_transit', 'odd_even_mismatch', 'transit_times',
+            'per_transit_count', 'transit_count', 'distinct_transit_count',
+            'empty_transit_count', 'FAP', 'in_transit_count', 'after_transit_count',
+            'before_transit_count', 'periods', 'power', 'power_raw', 'SR', 'chi2',
+            'chi2red', 'model_lightcurve_time', 'model_lightcurve_model',
+            'model_folded_phase', 'folded_y', 'folded_dy', 'folded_phase',
+            'model_folded_model'])
+
+        The descriptions are here:
+        https://transitleastsquares.readthedocs.io/en/latest/Python%20interface.html#return-values
+
+        The remaining resultdict is:
+
+            resultdict = {
+                'tlsresult':tlsresult,
+                'bestperiod': the best period value in the periodogram,
+                'bestlspval': the periodogram peak associated with the best period,
+                'nbestpeaks': the input value of nbestpeaks,
+                'nbestlspvals': nbestpeaks-size list of best period peak values,
+                'nbestperiods': nbestpeaks-size list of best periods,
+                'lspvals': the full array of periodogram powers,
+                'periods': the full array of periods considered,
+                'tlsresult': Astropy tls result object (BoxLeastSquaresResult),
+                'tlsmodel': Astropy tls BoxLeastSquares object used for work,
+                'method':'tls' -> the name of the period-finder method,
+                'kwargs':{ dict of all of the input kwargs for record-keeping}
+            }
+    """
+
+    if not magsarefluxes:
+        errmsg = ('IDK if transitleastsquares supports magnitudes. '
```

Comment:
i do know (after testing). it seems not to!

Reply:
Would be good to convert input mags to relative (normalized to 1) fluxes like you do for the test here so magsarefluxes works as expected for both mags and fluxes.